Adding CSV reader (Singleton) to read test data from input files

tcsako opened the pull request:
CSVReaderUtilitySingleton - CSV reader, using Singleton design pattern
CSVTestInput - POJO file to use test inputs in scripts
test_data_rest.csv - input data examples
pom.xml - add OpenCSV lib as dependency

Usage:
List<CSVTestInput> testInputs = CSVReaderUtilitySingleton.getInstance().getIntput(null);

Commit:
Adding CSV reader (Singleton) to read test data from input files

jozsefkoza commented on example-tests/pom.xml:
```diff
@@ -1,39 +1,44 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Please check your IDE formatter. This formatting was unnecessary.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java:
```diff
@@ -0,0 +1,105 @@
+			//TODO: use Log4j for logging
+			e.printStackTrace();
+			return null;
+		} catch (IOException e) {
```
You can handle multiple exceptions in one catch block,

jozsefkoza commented on example-tests/pom.xml:
```diff
@@ -1,39 +1,44 @@
+			<groupId>org.seleniumhq.selenium</groupId>
+			<artifactId>selenium-java</artifactId>
+		</dependency>
+		<dependency>
```
Please add this to dependencyManagement section. Also, consider using

```
<dependency>
    <groupId>org.apache.commons</groupId>
    <artifactId>commons-csv</artifactId>
    <version>1.2</version>
</dependency>
```

as it supports header lines by design.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java:
```diff
@@ -0,0 +1,105 @@
+	 * @return list of input data in CSVTestInput format
+	 */
+	@SuppressWarnings("resource")
+	public List<CSVTestInput> getIntput(String fileName) {
```
This is too specific. It should be either generic or return a list of Strings for now.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java:
```diff
@@ -0,0 +1,105 @@
+ * @author Tamas_Csako
+ *
+ */
+public class CSVReaderUtilitySingleton {
```
It's unnecessary to be singleton. Its instantiation cost is nearly nothing, the costly file reading is defined in an instance method.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java:
```diff
@@ -0,0 +1,105 @@
+		
+		//Read all input data from resource file
+		try {
+			CSVReader reader = new CSVReader(new FileReader(fileName));
```
Should define CSVReader in the try block definition, since it implements AutoClosable:

```
try (CSVReader reader = ...) {
  reader.readAll();
}
```

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/pojo/csv/CSVTestInput.java:
```diff
@@ -0,0 +1,47 @@
+ * @author Tamas_Csako
+ *
+ */
+public class CSVTestInput {
```
Consider creating a builder for this class.

Commit:
Updated CSV reader

Details:
- Add private exception to handle issues during reading from CSV file
- Use apache's CSV reader (it is more flexible) in CSV reader to allow usage of comments in CSV input files
- add header in CSV input file
- move related dependencies into parent pom.xml
- implement builder pattern in CSVRestTestInput POJO
- update test class accordingly

Commit:
- Clean up test class
- Remove second line in CSV

Commit:
Update test script to use only one line from the CSV input file, also clean up code in test class

Commit:
Remove unused idiot comment :)

jozsefkoza commented:
There are conflicts between master and the actual branch. I will resolve it, next time please do it on your own.

jozsefkoza committed:
Merge branch 'master' into cst

Conflicts:
	parent/pom.xml
